apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/Operations.java

```diff
@@ -201,22 +200,18 @@ public void deleteStagedOrphanDirectory(
     }
   }
 
-  /** Run ExpireSnapshots operation on a given fully-qualified table name. */
-  public ExpireSnapshots.Result expireSnapshots(String fqtn, long expireBeforeTimestampMs) {
-    return expireSnapshots(getTable(fqtn), expireBeforeTimestampMs);
+  /** Expire snapshots on a given fully-qualified table name. */
+  public void expireSnapshots(String fqtn, long expireBeforeTimestampMs) {
+    expireSnapshots(getTable(fqtn), expireBeforeTimestampMs);
   }
 
-  /** Run ExpireSnapshots operation on a given {@link Table}. */
-  public ExpireSnapshots.Result expireSnapshots(Table table, long expireBeforeTimestampMs) {
-    return SparkActions.get(spark)
-        .expireSnapshots(table)
+  /** Expire snapshots on a given {@link Table}. */
+  public void expireSnapshots(Table table, long expireBeforeTimestampMs) {
+    table
+        .expireSnapshots()
+        .cleanExpiredFiles(false)
         .expireOlderThan(expireBeforeTimestampMs)
-        .deleteWith(
```

Comment:
Interesting find. How did you catch this? What is the default behavior if delete functor is not supplied? How was this verified to  be leaner?

Reply:
Currently `SparkActions.get(spark).expireSnapshots()` returns [BaseExpireSnapshotsSparkAction](https://github.com/apache/iceberg/blob/e340ad5be04e902398c576f431810c3dfa4fe717/spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/actions/BaseExpireSnapshotsSparkAction.java). That object executes the action, that's when [expire](https://github.com/apache/iceberg/blob/e340ad5be04e902398c576f431810c3dfa4fe717/spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/actions/BaseExpireSnapshotsSparkAction.java#L158) is called. There it does normal metadata commit with all filters applied, and then constructs dataset for orphan files that it will materialize. So materialization part is expensive and this change removes that.

[Default delete](https://github.com/apache/iceberg/blob/e340ad5be04e902398c576f431810c3dfa4fe717/spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/actions/BaseExpireSnapshotsSparkAction.java#L84) is to delete files.

Reply:
https://github.com/apache/iceberg/blob/e340ad5be04e902398c576f431810c3dfa4fe717/spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/actions/BaseExpireSnapshotsSparkAction.java#L148-L156
Per the comment expire() is materializing is preparing dataset for orphan files. If we are calling that method I don't get how this change removes the materialization.

Reply:
The call chain is this: execute -> doExecute -> expire -> deleteFiles